master/internal/resourcemanagers/agent.go

```diff
@@ -11,9 +11,10 @@ import (
 // agentState holds the scheduler state for an agent. The implementation of agent-related operations
 // (e.g., socket I/O) is deferred to the actor.
 type agentState struct {
-	handler *actor.Ref
-	devices map[device.Device]*cproto.ID
-	label   string
+	handler      *actor.Ref
+	devices      map[device.Device]*cproto.ID
+	label        string
+	resourcePool string
```

Comment:
blocking: resource pool shouldn't be a field since all the `agentState` in one resource pool belongs to the resource pool

Reply:
Fixed